modules/runner/src/integrationTest/java/org/apache/ignite/internal/table/ItTableScanTest.java

```diff
@@ -789,8 +789,6 @@ public void testCompositeScanRequest(int requestAmount1, int requestAmount2, boo
         if (tx != null) {
             tx.rollback();
         }
-
-        assertThat(scanned, willCompleteSuccessfully());
```

Comment:
According to the `Subscription#cancel` javadocs:

```
/**
 * Causes the Subscriber to (eventually) stop receiving
 * messages.  Implementation is best-effort -- additional
 * messages may be received after invoking this method.
 * A cancelled subscription need not ever receive an
 * {@code onComplete} or {@code onError} signal.
 */
public void cancel();
```

Since in this test we cancel the subscription, we cannot expect that `onComplete` (which completes the future) will be called

Reply:
Do you also mean that there are more rows in partition than requested in both first and second requests?

Reply:
yes